pyccel/codegen/printing/cucode.py

```diff
@@ -0,0 +1,153 @@
+    "stdlib",
+    "string",
+    "tgmath",
+    "inttypes",
+)
+
+errors = Errors()
+
+# TODO: add examples
+
+__all__ = ["CudaCodePrinter", "cucode"]
+
+class CudaCodePrinter(CCodePrinter):
+    """
+    A printer for printing code in Cuda.
+
+    A printer to convert Pyccel's AST to strings of cuda code.
+    As for all printers the navigation of this file is done via _print_X
+    functions.
+
+    Parameters
+    ----------
+    filename : str
+            The name of the file being pyccelised.
+    prefix_module : str
+            A prefix to be added to the name of the module.
+    """
+    printmethod = "_cucode"
+    language = "cuda"
+
+    def __init__(self, filename, prefix_module = None):
+
+        errors.set_target(filename, 'file')
+
+        super().__init__(filename)
+        self.prefix_module = prefix_module
+        self._additional_imports = {'stdlib':c_imports['stdlib']}
+        self._additional_code = ''
+        self._additional_args = []
+        self._temporary_args = []
+        self._current_module = None
+        self._in_header = False
+
+    def _print_Module(self, expr):
+        self.set_scope(expr.scope)
+        self._current_module = expr.name
+        body    = ''.join(self._print(i) for i in expr.body)
+
+        global_variables = ''.join([self._print(d) for d in expr.declarations])
+
+        # Print imports last to be sure that all additional_imports have been collected
+        imports = [Import(expr.name, Module(expr.name,(),())), *self._additional_imports.values()]
+        c_headers_imports = ''
+        local_imports = ''
+
+        for imp in imports:
+            if imp.source in c_library_headers:
+                c_headers_imports += self._print(imp)
+            else:
+                local_imports += self._print(imp)
+
+        imports = f'{c_headers_imports}\
+                    extern "C"{{\n\
+                    {local_imports}\
+                    }}'
+
+        code = f'{imports}\n\
+                 {global_variables}\n\
+                 {body}\n'
+
+        self.exit_scope()
+        return code
+
+    def _print_Declare(self, expr):
+        if isinstance(expr.variable, InhomogeneousTupleVariable):
+            return ''.join(self._print_Declare(Declare(v.dtype,v,intent=expr.intent, static=expr.static)) for v in expr.variable)
+
+        declaration_type = self.get_declare_type(expr.variable)
+        variable = self._print(expr.variable.name)
+
+        if expr.variable.is_stack_array:
+            preface, init = self._init_stack_array(expr.variable,)
+        elif declaration_type == 't_ndarray' and not self._in_header:
+            preface = ''
+            init    = ' = {.shape = NULL}'
+        else:
+            preface = ''
+            init    = ''
+
+        declaration = f'{declaration_type} {variable}{init};\n'
+
+        return preface + declaration
+
+def cucode(expr, filename, assign_to=None, **settings):
+    """
+    Converts an expr to a string of cuda code.
```

Comment:
```suggestion
    Convert an expr to a string of cuda code.
```

Is this function used?

Reply:
what do you mean by used?

Reply:
removed in [6dff5bb](https://github.com/pyccel/pyccel-cuda/pull/32/commits/6dff5bbdd854006d5d6ee682a8bee3e98bbed5f3)